llvm/lib/Target/PowerPC/PPCInstrVSX.td

```diff
@@ -2707,6 +2707,44 @@ def : Pat<(vselect v4i32:$vA, v4f32:$vB, v4f32:$vC),
           (XXSEL $vC, $vB, $vA)>;
 def : Pat<(vselect v2i64:$vA, v2f64:$vB, v2f64:$vC),
           (XXSEL $vC, $vB, $vA)>;
+
+def : Pat<(vselect (v4i32 (setcc v4i32:$lhs, v4i32:$rhs, SETEQ)),
+                    v4i32:$tval, v4i32:$fval),
+          (XXSEL $fval, $tval, (XVCMPEQDP $lhs, $rhs))>;
+def : Pat<(vselect (v4i32 (setcc v4i32:$lhs, v4i32:$rhs, SETNE)),
+                    v4i32:$tval, v4i32:$fval),
+          (XXSEL $tval, $fval, (XVCMPEQDP $lhs, $rhs))>;
+def : Pat<(vselect (v4i32 (setcc v4i32:$lhs, v4i32:$rhs, SETGE)),
+                    v4i32:$tval, v4i32:$fval),
+          (XXSEL $fval, $tval, (XVCMPGEDP $lhs, $rhs))>;
```

Comment:
Can you explain why it is OK to use double precision floating point compares to compare integers?